oauthlib/openid/connect/core/grant_types/hybrid.py

```diff
@@ -35,6 +35,9 @@ def __init__(self, request_validator=None, **kwargs):
         self.register_code_modifier(self.add_id_token)
         self.register_token_modifier(self.add_id_token)
 
+    def add_id_token(self, token, token_handler, request):
+        return super().add_id_token(token, token_handler, request, nonce=request.nonce)
```

Comment:
is this nonce related to this? https://stackoverflow.com/questions/3690738/need-help-understanding-nonce

Reply:
Yep, or better description is [in the spec](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest). Its used to prevent replay attacks, the client generates a random nonce and stores it in a session, and then compares the nonce in the response with the nonce in the session. If it is missing either in the session or the response, or does not match, the response is invalid.

Its a required parameter for the implicit flow, a sometimes required parameter for the hybrid flow and an optional parameter for authorization code flow.

Handling it, however, is not optional - if a nonce is presented in the authorization endpoint, it must be in the ID token or a client should reject the response as invalid.